features/conversation-list/hooks/use-conversation-is-unread.ts

```diff
@@ -1,16 +1,68 @@
+import { conversationIsUnreadByTimestamp } from "@/features/conversation/utils/conversation-is-unread-by-current-account";
+import { getConversationDataQueryOptions } from "@/queries/use-conversation-data-query";
+import { getConversationQueryOptions } from "@/queries/useConversationQuery";
+import { useCurrentAccount } from "@data/store/accountsStore";
+import { useQuery } from "@tanstack/react-query";
 import { ConversationTopic } from "@xmtp/react-native-sdk";
-import { useConversationByTopic } from "./use-conversation-by-topic";
-import { useConversationIsUnread } from "./useMessageIsUnread";
+import { useMemo } from "react";
 
-export const useConversationIsUnreadByTopic = ({
-  topic,
-}: {
+type UseConversationIsUnreadArgs = {
   topic: ConversationTopic;
-}) => {
-  const conversation = useConversationByTopic(topic);
-  return useConversationIsUnread({
-    topic,
-    lastMessage: conversation?.lastMessage,
-    timestampNs: conversation?.lastMessage?.sentNs ?? 0,
+};
+
+export const useConversationIsUnread = ({
+  topic,
+}: UseConversationIsUnreadArgs) => {
+  const currentAccount = useCurrentAccount();
+  // const { data: currentUserInboxId } = useCurrentAccountInboxId();
+
+  const { data: conversationData, isLoading: isLoadingConversationData } =
+    useQuery(
+      getConversationDataQueryOptions({
+        account: currentAccount!,
+        topic,
+        context: "useConversationIsUnread",
+      })
+    );
+
+  const { data: lastMessage, isLoading: isLoadingLastMessage } = useQuery({
```

Comment:
Just a rq question from me: if we already have `conversationData`, when will `lastMessage` be different than `conversationData.lastMessage`? In other words why have two `useQuery` here?